src/regexp.rs

```diff
@@ -4,10 +4,6 @@ use regex::{Regex, RegexBuilder};
 // is anchored and has capturing groups for name, tag, and digest components.
 pub const REFERENCE_REGEXP: &str = r"^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$";
 
-/// ANCHORED_NAME_REGEXP is used to parse a name value, capturing the domain and
-/// trailing components.
-pub const ANCHORED_NAME_REGEXP: &str = r"^(?:((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?)/)?([a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)$";
```

Comment:
We did have this publicly exposed. I assume we aren't using it anywhere? If not, it is probably ok to remove, but we'll want to document the breaking change in the next release

Reply:
It was used before we merged https://github.com/krustlet/oci-distribution/pull/5 . I didn't think about end users relying on that. I can leave that and just suppress this specific clippy warning.

What do you prefer?

Reply:
I think leaving it and suppressing the clippy warning here is the best option

Reply:
Done